modules/shadertools/src/modules/lighting/pbr-material/pbr-material.ts

```diff
@@ -93,25 +62,29 @@ export type PBRMaterialUniforms = {
   scaleFGDSpec: Readonly<Vector4 | NumberArray4>;
 };
 
+export type PBRMaterialProps = PBRMaterialBindings & PBRMaterialUniforms;
+
 /**
  * An implementation of PBR (Physically-Based Rendering).
  * Physically Based Shading of a microfacet surface defined by a glTF material.
  */
 export const pbrMaterial: ShaderModule<PBRMaterialProps, PBRMaterialUniforms> = {
-  name: 'pbr',
+  name: 'pbrMaterial',
   vs,
   fs,
   defines: {
-    LIGHTING_FRAGMENT: 1,
-    HAS_NORMALMAP: 0,
-    HAS_EMISSIVEMAP: 0,
-    HAS_OCCLUSIONMAP: 0,
-    HAS_BASECOLORMAP: 0,
-    HAS_METALROUGHNESSMAP: 0,
-    ALPHA_CUTOFF: 0,
-    USE_IBL: 0,
-    PBR_DEBUG: 0
+    LIGHTING_FRAGMENT: 1
+    // TODO defining these as 0 breaks shader
```

Comment:
Interesting. Some thoughts:

-We may want to separate in the API between flags (for code enablement, which are either defined or not) and constants (which would inject actual values). 
- WGSL has built-in support for constants (that does not require recompiling the shader!), but not defines. However shadertools has added a simple #ifdef preprocessor system.

So, `props.defines` could just be an array or a Set, or a Record<string, boolean>. and `props.constants` a map of numeric values

Reply:
Interesting idea. `#ifdef NAME` is generally a bit frustrating as setting it to `0` evaluates to true in the preprocessor. Some thoughts:

- Restricting them to boolean values makes the name `props.defines` sound weird, perhaps `props.flags`?
- Could our preprocessor instead use a custom syntax? E.g. 
  ```
  // if FLAG(FLAG_NAME)
  ...
  // else
  ...
  // endif
  ``` 
- We do things like `#define SHADER_NAME simple-mesh-layer-vs`, it is weird to restrict defines from the outside to be boolean, and then sometimes use strings